packages/drivers/routerlicious-driver/src/retriableGitManager.ts

```diff
@@ -0,0 +1,155 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type * as git from "@fluidframework/gitresources";
+import type * as protocol from "@fluidframework/protocol-definitions";
+import { IGitManager, IWholeSummaryPayload, IWriteSummaryResponse } from "@fluidframework/server-services-client";
+import { runWithRetry } from "@fluidframework/driver-utils";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+
+export class RetriableGitManager implements IGitManager {
+    constructor(
+        private readonly internalGitManager: IGitManager,
+        private readonly logger: ITelemetryLogger,
+    ) {
+    }
+
+    public async getHeader(id: string, sha: string): Promise<protocol.ISnapshotTree> {
+        return this.runWithRetry(
+            async () => this.internalGitManager.getHeader(id, sha),
+            "gitManager_getHeader",
```

Comment:
I know we use this pattern in other places, but I think it's not worth recording the name of method in telemetry.
Collecting stack should be enough as it will contain public method names (as they are not affected by minimization and bundling). At the same time name alone (without stack) is not sufficient for most of the investigations, as we want to know the caller.

We need to validate these claims, but I feel we increase bundle sizes for nothing.

Reply:
Opened #6518